proof/encoding.go

```diff
@@ -37,6 +40,10 @@ func BlockHeaderEncoder(w io.Writer, val any, buf *[8]byte) error {
 }
 
 func BlockHeaderDecoder(r io.Reader, val any, buf *[8]byte, l uint64) error {
+	if l > BlockHeaderSizeBytes {
```

Comment:
If block headers are fixed-size, why is this set up for variable length at all?

Reply:
It is a static size record. But you're right, we can rewrite it somewhat to be more static.